### PR TITLE
test: convert Google Gemini tests to VCR

### PR DESCRIPTION
**Pull request by codefromthecrypt**
part of https://github.com/square/exchange/issues/67 closes https://github.com/square/exchange/pull/71

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@baxen I'll move each pending PR from exchange to here one-by-one, here's the first!

File: packages/exchange/tests/providers/conftest.py [hunk 1]
```diff
@@ -128,4 +149,4 @@ def vision(provider_cls: Type[Provider], model: str, **kwargs) -> Tuple[Message,
             content=[ToolResult(tool_use_id="xyz", output='"image:tests/test_image.png"')],
         ),
     ]
-    return provider.complete(model=model, system=system, messages=messages, tools=None, **kwargs)
+    return provider.complete(model=model, system=system, messages=messages, tools=(), **kwargs)
```
Comment (codefromthecrypt) [inline packages/exchange/tests/providers/conftest.py:152]: There are a lot of IDE warnings about call sites we use, as it is at the moment `tools: Tuple[Tool]`. So, this clears warnings about passing None. Though I wonder if it shouldn't be `tools: Tuple[Tool, ...]`? Seems that would be more explicit about possibly none, and also clear some other warnings. If you feel this is worthwhile I can in a follow-up or different PR.

Screenshot 2024-10-05 at 7 32 24 AM

**Comment by lamchau** [comment]
:nit: could we format the json so it's easier to read?

```yaml
        string: |-
          {
            "candidates": [
              {
                "content": { ... }
               }
```

**Comment by codefromthecrypt** [comment]
@lamchau 

> :nit: could we format the json so it's easier to read?

technically yes, it will require some engineering to write a custom serializer as vcr has no options built-in. I'm not opposed to this, but it also will affect the openai data already here, and could be done as a separate task. Is it ok if we decouple this? You can assign that issue to me which I can try on a weekend or a flight.

**Comment by lamchau** [comment]
@codefromthecrypt sure! just a nit - i just like to know what's being passed in

just for learning: is the custom serializer required to dump the request/response to yaml? never used it before so i was naively thinking using a `yq` transform for post-processing - the serializer sounds much more robust (and involved)

edit: made issue #127, can't seem to assign it - thanks for taking a look!

**Review by michaelneale — APPROVED** [review]

**Review by baxen — APPROVED** [review]
Looks great!

**Comment by codefromthecrypt** [comment]
@lamchau 

> just for learning: is the custom serializer required to dump the request/response to yaml? never used it before so i was naively thinking using a yq transform for post-processing - the serializer sounds much more robust (and involved)

So, I didn't consider yq (like something we could do in `just format`), it is possible that doing that wouldn't taint the input expectations vcrpy and look like request drift. That could be the easier way out vs a custom serializer, which is the current suggestion from the project. https://github.com/kevin1024/vcrpy/issues/580

File: packages/exchange/tests/providers/conftest.py (hunk 1 above)
Comment (codefromthecrypt) [reply packages/exchange/tests/providers/conftest.py:152]: https://github.com/block-open-source/goose/pull/149

**Pull request merged**